code/modules/crafting/table.dm

```diff
@@ -20,18 +20,18 @@
 			return 0
 	return 1
 
-/obj/structure/table/proc/check_table()
+/obj/structure/table/proc/check_table(datum/table_recipe/R)
 	table_contents = list()
 	for(var/obj/item/I in loc)
 		if(istype(I, /obj/item/stack))
 			var/obj/item/stack/S = I
 			table_contents[I.type] += S.amount
 		else
-			if(istype(I, /obj/item/weapon/reagent_containers))
-				for(var/datum/reagent/R in I.reagents.reagent_list)
-					table_contents[R.type] += R.volume
-
 			table_contents[I.type] += 1
+	for(var/obj/item/weapon/reagent_containers/RC in loc)
+		if((RC.flags & OPENCONTAINER) || (R && (RC.type in R.reqs)))
+			for(var/datum/reagent/A in RC.reagents.reagent_list)
+				table_contents[A.type] += A.volume
```

Comment:
1 - Why a separate loop, you can handle this all in one pass, instead you decide to loop through all items again just for this case
2 - this is the proc that is supposed to check all the things on the table, and it does it, yet for some reason you are trying to tie in a separate recipe here for seemingly no purpose